docs/ops/memory/mem_tuning.md

```diff
@@ -0,0 +1,87 @@
+
+<strong>Note:</strong> If you configure the *total Flink memory* Flink will implicitly add JVM memory components
+to derive the *total process memory* and request a container with the memory of that derived size,
+see also [detailed Memory Model](mem_setup.html#detailed-memory-model).
+
+<div class="alert alert-warning">
+  <strong>Warning:</strong> If Flink or user code allocates unmanaged off-heap (native) memory beyond the container size
+  the job can fail because the deployment environment can kill the offending containers.
+</div>
+See also description of [container memory exceeded](mem_trouble.html#container-memory-exceeded) failure.
+
+## Configure memory for state backends
+
+When deploying a Flink streaming application, the type of [state backend](../state/state_backends.html) used
+will dictate the optimal memory configurations of your cluster.
+
+### Heap state backend
+
+When running a stateless job or using a heap state backend ([MemoryStateBackend](../state/state_backends.html#the-memorystatebackend)
+or [FsStateBackend](../state/state_backends.html#the-fsstatebackend), set [managed memory](mem_setup.html#managed-memory) to zero.
+This will ensure that the maximum amount of memory is allocated for user code on the JVM.
+
+### RocksDB state backend
+
+The [RocksDBStateBackend](../state/state_backends.html#the-rocksdbstatebackend) uses native memory. By default,
+the RocksDB is setup to limit native memory allocation to the size of the [managed memory](mem_setup.html#managed-memory).
+Therefore, it is important to reserve enough *managed memory* for your state use case. If you disable the default RocksDB memory control,
+task executors can be killed in containerized deployments if RocksDB allocates memory above the limit of the requested container size
+(the [total process memory](mem_setup.html#configure-total-memory)).
+See also [state.backend.rocksdb.memory.managed](../config.html#state-backend-rocksdb-memory-managed).
+
+## Configure memory for batch jobs
+
+[Managed memory](../memory/mem_setup.html#managed-memory) helps Flink to run the batch operators efficiently.
+Therefore, the size of the [managed memory](mem_setup.html#managed-memory) can affect the performance of [batch jobs](../../dev/batch).
+If it makes sense, Flink will try to allocate and use as much *managed memory* as configured for batch jobs but not beyond its limit.
+It prevents OutOfMemoryExceptions because Flink knows how much memory it can use to execute operations.
+If Flink runs out of [managed memory](../memory/mem_setup.html#managed-memory), it utilizes disk space.
+Using [managed memory](../memory/mem_setup.html#managed-memory), some operations can be performed directly
+on the raw data without having to deserialize the data to convert it into Java objects. All in all,
+[managed memory](../memory/mem_setup.html#managed-memory) improves the robustness and speed of the system.
```

Comment:
> Flink's [batch](../../dev/batch) operators leverage [managed memory](../memory/mem_setup.html#managed-memory) to run more efficiently. In doing so, some operations can be performed directly on raw data without having to be deserialized into Java objects. This means that managed memory configurations have practical effects on the performance of your applications. Flink will attempt to allocate and use as much managed memory as configured for batch jobs but not go beyond its limits. This prevents `OutOfMemoryException`'s because Flink knows precisely how much memory it has to leverage. If the managed memory is not sufficient, Flink will gracefully spill to disk.